Source/control.cpp

```diff
@@ -310,6 +310,36 @@ void SetSpellTrans(spell_type t)
 	}
 }
 
+void PrintSBookSpellType(const Surface &out, Point position, const std::string &text, uint8_t rectColorIndex)
+{
+	Point rect {position};
+	rect += Displacement { 0, -SPLICONLENGTH + 1};
+
+	// Top
+	DrawHorizontalLine(out, rect, SPLICONLENGTH, rectColorIndex);
+	DrawHorizontalLine(out, rect + Displacement {0, 1}, SPLICONLENGTH, rectColorIndex);
+
+	// Bottom
+	DrawHorizontalLine(out, rect + Displacement {0, SPLICONLENGTH - 2}, SPLICONLENGTH, rectColorIndex);
+	DrawHorizontalLine(out, rect + Displacement {0, SPLICONLENGTH - 1}, SPLICONLENGTH, rectColorIndex);
+
+	// Left Side
+	DrawVerticalLine(out, rect, SPLICONLENGTH, rectColorIndex);
+	DrawVerticalLine(out, rect + Displacement {1, 0}, SPLICONLENGTH, rectColorIndex);
+
+	// Right Side
+	DrawVerticalLine(out, rect + Displacement {SPLICONLENGTH - 2, 0}, SPLICONLENGTH, rectColorIndex);
+	DrawVerticalLine(out, rect + Displacement {SPLICONLENGTH - 1, 0}, SPLICONLENGTH, rectColorIndex);
+
+	// Align the spell type text with bottom of spell icon
+	position += Displacement { SPLICONLENGTH / 2 - GetLineWidth(text.c_str()) / 2, -14 };

```

Comment:
Looks like this is still aligned for a 57x57 icon.
```suggestion
	position += Displacement { SPLICONLENGTH / 2 - GetLineWidth(text.c_str()) / 2, -15 };
```

Reply:
Thanks!